### DISCUSSION TITLE
Kafka pingBroker can hang on unreachable broker so please add a timeout.

### DISCUSSION THREAD

**Discussion by Adinadu** [minio/minio#17043] [Show and tell]
func (k KafkaArgs) pingBrokers() bool {
        timeout := time.Second * 60
        d := net.Dialer{Timeout: timeout}
        for _, broker := range k.Brokers {
                _, dErr := d.Dial("tcp", broker.String())
                if dErr == nil {
                        return true
                }
        }
        return false
}

**Comment by harshavardhana** [comment]
It's done in latest release

**Comment by Adinadu** [comment]
Awesome!
Thank you.
